generators/server/templates/_pom.xml

```diff
@@ -1357,6 +1357,44 @@
                     <artifactId>spring-boot-devtools</artifactId>
                     <optional>true</optional>
                 </dependency>
+                <%_ if (devDatabaseType === 'mysql' && prodDatabaseType !== 'mysql') { _%>
+                <dependency>
+                    <groupId>mysql</groupId>
+                    <artifactId>mysql-connector-java</artifactId>
+                </dependency>
+                <%_ } _%>
+                <%_ if (devDatabaseType === 'mariadb' && prodDatabaseType !== 'mariadb') { _%>
+                <dependency>
+                    <groupId>org.mariadb.jdbc</groupId>
+                    <artifactId>mariadb-java-client</artifactId>
+                </dependency>
+                <%_ } _%>
+                <%_ if (devDatabaseType === 'mssql' && prodDatabaseType !== 'mssql') { _%>
+                <dependency>
+                    <groupId>com.microsoft.sqlserver</groupId>
+                    <artifactId>mssql-jdbc</artifactId>
+                    <version>${mssql-jdbc.version}</version>
+                </dependency>
+                <dependency>
+                    <groupId>com.github.sabomichal</groupId>
+                    <artifactId>liquibase-mssql</artifactId>
+                    <version>${liquibase-mssql.version}</version>
+                </dependency>
+                <%_ } _%>
+                <%_ if (devDatabaseType === 'oracle' && prodDatabaseType !== 'oracle') { _%>
+                <!-- more information at https://blogs.oracle.com/dev2dev/entry/how_to_get_oracle_jdbc -->
+                <dependency>
+                    <groupId>com.oracle.jdbc</groupId>
+                    <artifactId>ojdbc7</artifactId>
+                    <version>${oracle.version}</version>
+                </dependency>
+                <%_ } _%>
+                <%_ if (devDatabaseType === 'postgresql' && prodDatabaseType !== 'postgresql') { _%>
```

Comment:
h2 seems missing

Reply:
Here it is https://github.com/erikkemperman/generator-jhipster/blob/881f02b653bdd1124510c149801e2711bb2952e9/generators/server/templates/_pom.xml#L315

I didn't make the same split for H2, because it is used with `test` scope in case it isn't the dev database. But of course I could do something similar.